### PR TITLE
polyml: enable tests

### PR DESCRIPTION
**Pull request by someplaceguy**
## Description of changes

Enable tests for polyml.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @maggesi @KoviRobi

### CODE REVIEW

File: pkgs/development/compilers/polyml/default.nix [hunk 1]
```diff
@@ -24,12 +31,9 @@ stdenv.mkDerivation rec {
...
 
-  src = fetchFromGitHub {
-    owner = "polyml";
-    repo = "polyml";
-    rev = "v${version}";
-    sha256 = "sha256-72wm8dt+Id59A5058mVE5P9TkXW5/LZRthZoxUustVA=";
-  };
+  doCheck = true;
+
+  checkPhase = "make check";
```
Comment (h7x4) [inline pkgs/development/compilers/polyml/default.nix:36]: You probably don't need to specify this. The default hook already checks for `make check` and `make test`, and runs whichever one it finds.

https://github.com/NixOS/nixpkgs/blob/bf76c9f706c23d8f39991a0c785fa882e0966064/pkgs/stdenv/generic/setup.sh#L1327-L1334

File: pkgs/development/compilers/polyml/default.nix [hunk 2]
```diff
@@ -10,6 +10,13 @@ stdenv.mkDerivation rec {
   pname = "polyml";
   version = "5.9.1";
 
+  src = fetchFromGitHub {
+    owner = "polyml";
+    repo = "polyml";
+    rev = "v${version}";
+    sha256 = "sha256-72wm8dt+Id59A5058mVE5P9TkXW5/LZRthZoxUustVA=";
+  };
+
```
Comment (h7x4) [inline pkgs/development/compilers/polyml/default.nix:19]: To keep the diff minimal, I suggest this either gets moved back to where it was, or at least put this in a separate commit stating you've moved it.

File: pkgs/development/compilers/polyml/default.nix (hunk 1 above)
Comment (someplaceguy) [reply pkgs/development/compilers/polyml/default.nix:36]: Indeed, initially I didn't add the `checkPhase = "make check"` line, but that didn't work for some reason. Instead, I got:
```
running tests
no Makefile or custom checkPhase, doing nothing
```

This is why I added that line.

File: pkgs/development/compilers/polyml/default.nix (hunk 2 above)
Comment (someplaceguy) [reply pkgs/development/compilers/polyml/default.nix:19]: Ok, I will split it into two commits.

**Comment by someplaceguy** [comment]
Split the commit into two, as requested by @h7x4. Thanks!

**Review by IogaMaster — APPROVED** [review]

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 264385` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>hol</li>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>polyml</li>
    <li>tlaps</li>
  </ul>
</details>

LGTM, just a nit

**Comment by someplaceguy** [comment]
@pbsds `checkTarget = "check";` didn't work (got the same `no Makefile or custom checkPhase, doing nothing` message and no tests were ran), but the other suggestion works fine.

I've updated the PR to incorporate the hash change in the master branch as well as @pbsds's suggestion.

Thanks!

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 264385` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>hol</li>
    <li>isabelle</li>
    <li>isabelle-components.isabelle-linter</li>
    <li>polyml</li>
    <li>tlaps</li>
  </ul>
</details>
